src/classes/User.ts

```diff
@@ -24,31 +36,45 @@ export default class User {
 
         if (this.#apiKey) {
             this.#authorizationHeader = {
-                headers: {
-                    Authorization: `Bearer ${this.#apiKey}`,
-                },
+                Authorization: `Bearer ${this.#apiKey}`,
             };
         } else {
             this.#authorizationHeader = undefined;
         }
+
+        this.#configuration = new Configuration({
+            apiKey: this.#apiKey,
+            fetchApi: fetch as unknown as ConfigurationParameters['fetchApi'],
+            headers: this.#authorizationHeader,
+            middleware: [],
+        });
     }
 
     /**
-     * Get a user's object using their user ID.
-     *
-     * @param {string} userID The Passage user ID
-     * @return {Promise<UserObject>} Passage User object
+     * Check if API key exists for this Passage instance
+     * @throws {PassageError} If the API key is missing.
      */
-    async get(userID: string): Promise<UserObject> {
+    private _apiKeyCheck(): void {
         if (!this.#apiKey) {
             throw new PassageError('A Passage API key is needed.');
         }
+    }
+
+    /**
+     * Get a user's object using their user ID.
+     *
+     * @param {string} userID The Passage user ID
+     * @return {Promise<UserInfo>} Passage User object
+     */
+    async get(userID: string): Promise<UserInfo> {
+        this._apiKeyCheck();
 
         try {
-            const response = await fetch({
-                ...this.#authorizationHeader,
-                method: 'GET',
-                url: `https://api.passage.id/v1/apps/${this.#appID}/users/${userID}`,
+            const client = new UsersApi(this.#configuration);
+
+            const response = await client.getUser({
```

Comment:
What do you think about initializing the client in the constructor as well?

Then this could just be
```ts
const response = await this.#client.getUser({
```

Reply:
Makes sense. I will also rework these a bit to mimic our classes to the generated api files. Leaving our custom functions in the Passage class to create a better division.